cmd/zc_traverser_gcp.go

```diff
@@ -24,20 +24,20 @@ type gcpTraverser struct {
 	incrementEnumerationCounter enumerationCounterFunc
 }
 
-func (t *gcpTraverser) IsDirectory(isSource bool) bool {
+func (t *gcpTraverser) IsDirectory(isSource bool) (bool, error) {
 	//Identify whether directory or not syntactically
 	isDirDirect := !t.gcpURLParts.IsObjectSyntactically() && (t.gcpURLParts.IsDirectorySyntactically() || t.gcpURLParts.IsBucketSyntactically())
 	if !isSource {
-		return isDirDirect
+		return isDirDirect, nil
 	}
 	bkt := t.gcpClient.Bucket(t.gcpURLParts.BucketName)
 	obj := bkt.Object(t.gcpURLParts.ObjectKey)
 	//Directories do not have attributes and hence throw error
 	_, err := obj.Attrs(t.ctx)
 	if err == gcpUtils.ErrObjectNotExist {
```

Comment:
For any other error then Not_Exist we shall be returning error here.

Reply:
We can apply error related changes of other platforms in a separate PR. The IsDirectory changes and error handling in this PR primarily applies to blobs.